### PR TITLE
Run length encoding

### PR DESCRIPTION
**Pull request by jerr-it**
### Describe your change:
Added run length encoding (compression algorithm)

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by CaedenPH — APPROVED** [review]

File: compression/run_length_encoding.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    [('A', 1)]
+    >>> run_length_encode("AA")
+    [('A', 2)]
+    >>> run_length_encode("AAADDDDDDFFFCCCAAVVVV")
+    [('A', 3), ('D', 6), ('F', 3), ('C', 3), ('A', 2), ('V', 4)]
+    """
+    encoded = []
+    count = 1
+
+    for i in range(len(input)):
```
Comment (cclauss) [inline compression/run_length_encoding.py:19]: In Python, `range(len())` is almost always a sign that https://docs.python.org/3/library/functions.html#enumerate should be used instead.

File: compression/run_length_encoding.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        if i + 1 < len(input) and input[i] == input[i + 1]:
+            count += 1
+        else:
+            encoded.append((input[i], count))
+            count = 1
+
+    return encoded
+
+
+def run_length_decode(input: list) -> str:
```
Comment (cclauss) [inline compression/run_length_encoding.py:29]: https://docs.python.org/3/library/functions.html#input is a builtin function in Python so please pick a different variable name.

File: compression/run_length_encoding.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    'AA'
+    >>> run_length_decode([('A', 3), ('D', 6), ('F', 3), ('C', 3), ('A', 2), ('V', 4)])
+    'AAADDDDDDFFFCCCAAVVVV'
+    """
+    decoded = ""
+
+    for i in input:
+        decoded += i[0] * i[1]
+
+    return decoded
```
Comment (cclauss) [inline compression/run_length_encoding.py:46]:
```suggestion
    return "".join(char * length for char, length in input)
```

**Review by cclauss — APPROVED** [review]
Nice one!!

**Pull request merged**